index.js

```diff
@@ -134,7 +153,19 @@ Shopify.prototype.request = function request(uri, method, key, data, headers) {
     parseJson: this.options.parseJson,
     timeout: this.options.timeout,
     responseType: 'json',
-    retry: 0,
+    retry:
+      this.options.maxRetries > 0
+        ? {
+            limit: this.options.maxRetries,
+            // Don't clamp Shopify retry-after header values too low.
+            maxRetryAfter: Infinity,
+            calculateDelay: (retryObject) => {
+              return (
+                maybeRetryMS(retryObject.error) || retryObject.computedValue
+              );
```

Comment:
Can you please move this at the root level? The function does not depend on anything defined on this scope, so there is no need to redefine it every time.